core/utils/container_utils.hpp

```diff
@@ -105,10 +107,13 @@ class raw_block_vector_base : private util::noncopyable {
   IRS_FORCE_INLINE bool empty() const noexcept { return buffers_.empty(); }
 
   IRS_FORCE_INLINE void clear() noexcept {
+    size_t size{0};
     for (auto& buffer : buffers_) {
+      size += buffer.size;
       alloc_.deallocate(buffer.data, buffer.size);
     }
     buffers_.clear();
+    resource_manager_.Decrease(size);
```

Comment:
```suggestion
    resource_manager_.DecreaseChecked(size);
```
Or check buffers_.empty in the function begining